drivers/can/can_esp32_twai.c

```diff
@@ -113,14 +224,40 @@ const struct can_driver_api can_esp32_twai_driver_api = {
 	.recover = can_sja1000_recover,
 #endif /* !CONFIG_CAN_AUTO_BUS_OFF_RECOVERY */
 	.timing_min = CAN_SJA1000_TIMING_MIN_INITIALIZER,
+#ifdef CONFIG_SOC_ESP32
 	.timing_max = CAN_SJA1000_TIMING_MAX_INITIALIZER,
+#else
+	/* larger prescaler allowed for newer ESP32-series MCUs */
+	.timing_max = {
+		.sjw = 0x4,
+		.prop_seg = 0x0,
+		.phase_seg1 = 0x10,
+		.phase_seg2 = 0x8,
+		.prescaler = 0x2000,
+	}
+#endif /* CONFIG_SOC_ESP32 */
 };
 
-#define CAN_ESP32_TWAI_DT_CDR_INST_GET(inst)                                                       \
+#define CAN_SJA1000_DT_CDR_INST_GET(inst)                                                          \
```

Comment:
This is still ESP32 TWAI binding specific and should be named as such.

Reply:
That'll work too. And yes, please leave the SJA1000 definitions in place for other frontends to use.